src/ffi/ios.rs

```diff
@@ -2,41 +2,110 @@ use core_foundation::{
     base::TCFType,
     string::{CFString, CFStringRef},
 };
-use http::Request;
+use http::{Request, HeaderValue, HeaderMap, StatusCode};
 use libc::c_char;
+use tracing::{info, debug};
+use url::Url;
+use anyhow::Result;
+use serde::{Serialize, Deserialize};
+use http::header::HeaderName;
+use std::collections::HashMap;
 
 use crate::{client::Client, DirectoryCache};
+use std::{fs, io};
 
-fn setup_logger() {
-    let subscriber = tracing_fmt::FmtSubscriber::builder()
-        .with_max_level(tracing::Level::DEBUG)
-        .with_writer(std::io::stderr)
-        .finish();
-    tracing::subscriber::set_global_default(subscriber).expect("to be the only logger");
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ArtiRequest {
+    method: String,
+    url: String,
+    headers: HashMap<String, Vec<String>>,
+    body: Vec<u8>,
+    dict_dir: String,
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn call_tls_get(domain_cc: *const c_char) -> CFStringRef {
-    let domain = cstring_to_str(&domain_cc);
+pub unsafe extern "C" fn call_arti(
+    request_json: *const c_char) -> CFStringRef {
+    setup_logger();
+
+    let ret = match _call_arti(cstring_to_str(&request_json)) {
+        Ok(ret) => ret,
+        Err(e) => ReturnStruct::err(e.to_string()),
+    };
```

Comment:
`unwrap_or_else` returns the `Ok` value or execute a function when it's an `Err` before returning.

```suggestion
    let ret = _call_arti(cstring_to_str(&request_json))
        .unwrap_or_else(|e| ReturnStruct::err(e.to_string()));
```